base_layer/core/src/transactions/transaction_components/encrypted_data.rs

```diff
@@ -58,17 +64,81 @@ const SIZE_NONCE: usize = size_of::<XNonce>();
 const SIZE_VALUE: usize = size_of::<u64>();
 const SIZE_MASK: usize = PrivateKey::KEY_LEN;
 const SIZE_TAG: usize = size_of::<Tag>();
-const SIZE_TOTAL: usize = SIZE_NONCE + SIZE_VALUE + SIZE_MASK + SIZE_TAG;
+pub const STATIC_ENCRYPTED_DATA_SIZE_TOTAL: usize = SIZE_NONCE + SIZE_VALUE + SIZE_MASK + SIZE_TAG;
 
 // Number of hex characters of encrypted data to display on each side of ellipsis when truncating
 const DISPLAY_CUTOFF: usize = 16;
 
-#[derive(
-    Debug, Copy, Clone, Deserialize, Serialize, PartialEq, Eq, Hash, BorshSerialize, BorshDeserialize, Zeroize,
-)]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash, BorshSerialize, BorshDeserialize, Zeroize)]
 pub struct EncryptedData {
     #[serde(with = "tari_utilities::serde::hex")]
-    data: [u8; SIZE_TOTAL], // nonce, encrypted value, encrypted mask, tag
+    data: Vec<u8>,
+}
+
+#[derive(Debug, Clone, Copy, Deserialize, Serialize, PartialEq, Eq)]
+pub enum PaymentId {
+    Zero,
+    U32(u32),
+    U64(u64),
+    U256(U256),
+}
```

Comment:
I think the brackets are a good idea instead of a perfect variable length.
- I propose these brackets:
  - 8 bytes
  - 16 bytes
  - 32 bytes
  - 16-byte increments up to 256 bytes (48, 64, 80, ..., 256)